File: core/src/main/java/com/shatteredpixel/shatteredpixeldungeon/ui/changelist/v0_X_Changes.java
package com.shatteredpixel.shatteredpixeldungeon.ui.changelist;

import com.shatteredpixel.shatteredpixeldungeon.Assets;
import com.shatteredpixel.shatteredpixeldungeon.messages.Messages;
import com.shatteredpixel.shatteredpixeldungeon.scenes.ChangesScene;
import com.shatteredpixel.shatteredpixeldungeon.sprites.CharSprite;
import com.shatteredpixel.shatteredpixeldungeon.ui.Icons;
import com.shatteredpixel.shatteredpixeldungeon.ui.Window;
import com.watabou.noosa.Image;

import java.util.ArrayList;

public class v0_X_Changes {
    public static void addAllChanges ( ArrayList<ChangeInfo> changeInfos ){
        add_v0_1_Changes(changeInfos);
    }

    public static void add_v0_1_Changes(ArrayList<ChangeInfo> changeInfos) {
        ChangeInfo changes = new ChangeInfo("v0.1", true, "");
        changes.hardlight( Window.TITLE_COLOR );
        changeInfos.add(changes);

        changes = new ChangeInfo(Messages.get(ChangesScene.class, "new"), false, null);
        changes.hardlight( Window.TITLE_COLOR );
        changeInfos.add(changes);

        changes.addButton( new ChangeButton(Icons.get(Icons.SHPX), "Developer Commentary",
                "_-_ Released MONTH DAY, 2020\n" +
                        "_-_ FIrst release!\n" +
                        "\n" +
                        "What is dev commentary?"));

        changes.addButton( new ChangeButton(new Image(Assets.WARRIOR, 0, 15, 12, 15), "New Leader class!",
                "The Leader is a new ally-centric class with a unique companion whom you must protect. " +
                        "She may be a burden in the beginning, but by the time you get to the Leader's subclasses you'll be glad you brought her along."));

        changes.addButton( new ChangeButton(new Image(Assets.SPECKS, 65, 0, 4, 5), "New music!",
                "A total of five new music tracks have been added: four dungeon tracks and a boss track. This means that each chapter now has its own theme."));

        changes = new ChangeInfo(Messages.get(ChangesScene.class, "changes"), false, null);
        changes.hardlight( CharSprite.WARNING );
        changeInfos.add(changes);

        changes.addButton( new ChangeButton(Icons.get(Icons.PREFS), Messages.get(ChangesScene.class, "misc"),
                "To help offset the massive file size increase brought by the new music, all existing compatibility code has been removed."));
    }
}

File: core/src/main/kotlin/com/right/helveticpixeldungeon/scenes/ui/changelist/V0_X_Changes.kt
package com.right.helveticpixeldungeon.scenes.ui.changelist

import com.shatteredpixel.shatteredpixeldungeon.sprites.ItemSprite
import com.shatteredpixel.shatteredpixeldungeon.ui.Icons
import com.shatteredpixel.shatteredpixeldungeon.ui.Window
import com.shatteredpixel.shatteredpixeldungeon.ui.changelist.ChangeButton
import com.shatteredpixel.shatteredpixeldungeon.ui.changelist.ChangeInfo

@Suppress("ClassName")
object V0_X_Changes {

    fun addAllChanges(changeInfos: ArrayList<ChangeInfo>) {
        add_v0_0(changeInfos)
    }
    fun add_v0_0(changeInfos: ArrayList<ChangeInfo>) {
        var changes = ChangeInfo("v0.0", true, "")
        changes.hardlight(Window.TITLE_COLOR)
        changeInfos.add(changes)

//        changes = ChangeInfo("", false, null)
//        changes.hardlight(Window.TITLE_COLOR)
//        changeInfos.add(changes)

        changes = ChangeInfo("v0.0.1", false, null)
        changes.hardlight(Window.TITLE_COLOR)
        changeInfos.add(changes)
        changes.addButton(ChangeHPDButton(Icons.PREFS.get(),"基础设施",""" 
            **-**  基本支持kotlin开发
            **-**  整理一下TitleScene,将HPD与SPD的更新记录分离
            **-**  删除了决斗家
         
        """.trimIndent()))

    }
}